initialize_package.sh

```diff
@@ -0,0 +1,309 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+# Constants and help message.
+OLD_NAME="template_component_package"
+HELP_MESSAGE="Usage: $0 [--dry-run]
+This script initializes the package by prompting you to select the component type (Python or C++), specify whether it's a lifecycle component, and enter the new package name. 
+It then sets up the files and directories based on your requirements.
+
+Options:
+  -n, --dry-run    Display the actions without making any changes.
+  -h, --help       Show this help message."
+
+DRY_RUN=false
+
+# Process options.
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -n|--dry-run)
+      DRY_RUN=true
+      shift
+      ;;
+    -h|--help)
+      echo "${HELP_MESSAGE}"
+      exit 0
+      ;;
+    -*)
+      echo "Unknown option: $1" >&2
+      echo "${HELP_MESSAGE}"
+      exit 1
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+# Logging helper.
+log() {
+  echo "$1"
+}
+
+# Prompt until nonempty input is given.
+prompt_nonempty() {
+  local prompt_msg="$1"
+  local input
+  while true; do
+    read -p $'\e[3;36m'"${prompt_msg}"$'\e[0m ' input
+    if [[ -n "${input}" ]]; then
+      echo "${input}"
+      return
+    else
+      log "Input cannot be empty. Please try again."
+    fi
+  done
+}
+
+# Get script directory.
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Prompt for user inputs with color-coded, italicized questions.
+NEW_PACKAGE_NAME=$(prompt_nonempty "Enter the new component package name:")
+
+while true; do
+  read -p $'\e[3;36mIs this a Python or C++ component? (python/c++):\e[0m ' COMPONENT_TYPE
+  COMPONENT_TYPE=$(echo "${COMPONENT_TYPE}" | tr '[:upper:]' '[:lower:]')
+  if [[ "${COMPONENT_TYPE}" == "python" || "${COMPONENT_TYPE}" == "c++" ]]; then
```

Comment:
I agree with @bpapaspyros, it would be great to have defaults so you can just press enter and handle `y` `n` on top of `yes`/`no`.